controllers/v1/sessions.js

```diff
@@ -9,76 +9,152 @@
 const sessionsHelper = require("../../services/helper/sessions");
 
 module.exports = class Sessions {
-    
-    form() {
-        return new Promise(async (resolve,reject) => {
-            try {
-                const sessionsForm = await sessionsHelper.form();
-                return resolve(sessionsForm);
-            } catch(error) {
-                return reject(error);
-            }
-        })
-    }
 
-    update(req) {
-        return new Promise(async (resolve,reject) => {
-            try {
-                const sessionUpdated = 
-                await sessionsHelper.update(
-                    req.params._id,
-                    req.body
-                );
+    async update(req) {
+        try {
+            if (req.params.id) {
+
+                if (req.headers.timezone) {
+                    req.body['timeZone'] = req.headers.timezone;
+                }
+
+                const sessionUpdated =
+                    await sessionsHelper.update(
+                        req.params.id,
+                        req.body,
+                        req.decodedToken._id,
+                        req.method
+                    );
+
+                return sessionUpdated;
+            } else {
+                if (req.decodedToken.name) {
+                    req.body.mentorName = req.decodedToken.name;
+                }
+
+                if (req.headers.timezone) {
+                    req.body['timeZone'] = req.headers.timezone;
+                }
+
+                const sessionCreated =
+                    await sessionsHelper.create(
+                        req.body, req.decodedToken._id
+                    );
+
+                return sessionCreated;
 
-                return resolve(sessionUpdated);
-            } catch(error) {
-                return reject(error);
             }
-        })
+        } catch (error) {
+            return error;
+        }
     }
 
-    details(req) {
-        return new Promise(async (resolve,reject) => {
-            try {
-                const sessionDetails = 
+    async details(req) {
+        try {
+            const sessionDetails =
                 await sessionsHelper.details(
-                    req.params._id
+                    req.params.id,
+                    req.decodedToken._id
+                );
+            return sessionDetails;
+        } catch (error) {
+            return error;
+        }
+    }
+
+    async list(req) {
+        try {
+            const sessionDetails =
+                await sessionsHelper.list(
+                    req.decodedToken._id,
+                    req.pageNo,
+                    req.pageSize,
+                    req.searchText,
+                    req.query.status
```

Comment:
@aman-g202  Why we need status in list ?

Reply:
Because they show in ui the status as live

Reply:
@aman-g202  Can we connect for this? Not understood properly.

Reply:
Yes